check-mysql/lib/check-mysql.go

```diff
@@ -52,12 +62,60 @@ SubCommands:`)
 	ckr.Exit()
 }
 
-func newMySQL(m mysqlSetting) mysql.Conn {
+func newDB(m mysqlSetting) (*sql.DB, error) {
 	proto := "tcp"
 	target := fmt.Sprintf("%s:%s", m.Host, m.Port)
 	if m.Socket != "" {
 		proto = "unix"
 		target = m.Socket
 	}
-	return mysql.New(proto, "", target, m.User, m.Pass, "")
+	cfg := &mysql.Config{
+		User:                 m.User,
+		Passwd:               m.Pass,
+		Net:                  proto,
+		Addr:                 target,
+		AllowNativePasswords: true,
```

Comment:
Hmm, do we need to support `AllowOldPasswords` or such? 🤔 (ref: https://pkg.go.dev/github.com/go-sql-driver/mysql#Config )

Reply:
`old_password` is not supported because the protocol is too legacy (~2004)  (https://github.com/go-sql-driver/mysql/wiki/old_passwords).